doc/content/examples/example4.md

```diff
@@ -34,4 +31,70 @@ layer effectively simulates the interface between two elastic layers (see [fig:d
 
-!bibtex bibliography
+# Example 4b: Frictional contact element test (cyclic loading)
+
+This example is a more general representation of the three block contact problem demonstrated in Example 4a. The problem formulation is exactly the same as the thin layer I-soil contact element sandwiched betweeen the concrete block and the soil block, which uses linear elastic material (see [fig:deformedcontact]). In the input file shown below; the  Materials, Functions, BC, and Controls block will be explained in some detail which is specific to the soil-structure interaction problem.
+
+!listing examples/ex04/ex04b/2BlockFriction_Isoilunit.i
+
+For this example, the units defining the parameters are N, kg, and meter (m); and the material definition is created using the following input:
+
+!listing examples/ex04/ex04b/2BlockFriction_Isoilunit.i
+ start=Materials
+ end=Preconditioning
+
+where;
+
+- `soil type = 'user-defined'` specifies that the backbone curve will be defined by the user and will correspond to a Coulomb friction formulation.
+
+- `backbone curve = backbone_curveunit.csv` The backbone curve is used to define critical shear stress at the interface which is obtained by multiplying the user-defined friction coefficient with the normal stresses . The shear strain is obtained by dividing the frictional shear stress with the shear modulus of the concrete. The above mentioned stress-strain values are entered via a .csv file or can be defined directly. The initial shear modulus of the thin layer is the same as that of concrete. The motivation is to model the contact element as a set of springs which yields once the critical stresses are reached.
+
+- `initial_shear_modulus = '2.71e10'` specifies the small strain shear modulus (elastic property) of
+ the soil and is equivalent to the shear modulus of the concrete block. It should be noted that if we were to model an embedded foundation the shear modulus of the thin layer must match that of the surrounding soil strata.
+
+- `poissons_ratio = '0.15'` specifies the Poisson's ratio for the thin layer elements (same as concrete). The linear elastic properties are matched with concrete in order to ensure uniform scaling of stresses during failure.
+
+- `block = 1002` specifies that the material will be assigned to a mesh volume with block id = 1002.
+
+- `initial_stress = 'initial_ymid 0 0 0 initial_ymid 0 0 0 initial_zzmid` see [Getting Started](manuals/user/index.md). This is not needed if pressure dependency is set to true or when the initial stresses are unknown.
+
+- `density = '8792'` specifies the density of the thin layer which is the same as concrete.
+
+- `p_ref = '904918'` is the reference pressure at which the backbone curve is defined. The reference pressure is obtained by performing a gravity analysis and switching off the inertia kernel for the first few time steps. This helps us in avoiding unrealistic displacement/acceleration response.
```

Comment:
The reference pressure is the hydrostatic stress state. Its not just the normal stress since there will be lateral stresses as well as obtained from the gravity analysis.  (If the boundaries were to move together by applying periodic boundary condition, then the stresses would result from the isotropic elastic equations)